source/common/umf_pools/disjoint_pool.cpp

```diff
@@ -894,6 +894,11 @@ umf_result_t DisjointPool::initialize(umf_memory_provider_handle_t *providers,
     if (numProviders != 1 || !providers[0]) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
+    // MinBucketSize parameter must be a power of 2.
+    if (!parameters.MinBucketSize ||
+        !((parameters.MinBucketSize & (parameters.MinBucketSize - 1)) == 0)) {
```

Comment:
nit: Can't it be a modulo operation? It's simpler:
```
if (!parameters.MinBucketSize || parameters.MinBucketSize % 2 != 0)
```

Reply:
modulo is more expensive.